### PR TITLE
Use ServiceWeakToken with asynchronous tasks

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR validation:

All framework unit tests pass.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@dan131riley this is the implementation I finally decided upon, replacing the one to which I pointed you.

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

File: FWCore/Services/plugins/InitRootHandlers.cc [hunk 1]
```diff
@@ -771,6 +771,11 @@ namespace edm {
 
       if (not threadTracker_) {
         threadTracker_ = std::make_unique<ThreadTracker>();
+        iReg.watchPostEndJob([]() {
+          if (threadTracker_) {
```
Comment (makortel) [inline FWCore/Services/plugins/InitRootHandlers.cc:775]: I'm wondering how `this` gets captured with empty `[]`.

Comment (Dr15Jones) [reply FWCore/Services/plugins/InitRootHandlers.cc:775]: weird indeed

Comment (Dr15Jones) [reply FWCore/Services/plugins/InitRootHandlers.cc:775]: Is threadTracker_ a `static`?

Comment (makortel) [reply FWCore/Services/plugins/InitRootHandlers.cc:775]: @Dr15Jones pointed out that `threadTracker_` is static.

Comment (makortel) [reply FWCore/Services/plugins/InitRootHandlers.cc:775]: Thanks

**Comment by makortel** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**